v2/pkg/engine/resolve/loader.go

```diff
@@ -1395,13 +1395,13 @@ func (l *Loader) executeSourceLoad(ctx context.Context, source DataSource, input
 
 		// Prevent that the context is destroyed when the loader hook return an empty context
 		if res.loaderHookContext != nil {
-			res.err = source.Load(res.loaderHookContext, input, res.out)
+			res.err = source.Load(res.loaderHookContext, input, l.ctx.Files, res.out)
```

Comment:
Here, we could call source.LoadWithFiles in case the source supports files and the client supplied files in the ctx, otherwise we just do source.Load. WDYT?

Reply:
Done